Mixmatch/streetview_dataset/parse_data_to_tfrecord_lib_test.ipynb

```diff
@@ -0,0 +1,239 @@
+    "## Test batch_read_write_tfrecords()\n",
+    "This file reads input tfrecords in batches, and process the bboxes that meet the conditions. And write back the labeles, and cropped images to a new tfrecord file.\n",
+    "\n",
+    "The batch_read_write_tfrecords also utilize read_tfrecord(), generate_tfexamples_from_detections(), write_tfexample_to_tfrecord(), parse_detection_confidences(), strip_top(all)_confidence_bbox(), img_to_example(), read_and_check_image()\n",
+    "\n",
+    "Therefore, this test is a functional test for all functions listed above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "INPUT_RECORD_DIR = './streetlearn-detections/'\n",
+    "INPUT_UCF_IMG_DIR = './UCF_Streetview_Dataset/raw/'\n",
+    "TF_FILE_DIR = './test_file.tfrecord'\n",
+    "writer = tf.io.TFRecordWriter(TF_FILE_DIR)\n",
+    "detection_property = {'include_top_camera':True, 'only_keep_top_confidence':True, 'balance':False}\n",
+    "file_range = [0, 1]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "batch_read_write_tfrecords(file_range, INPUT_RECORD_DIR, INPUT_UCF_IMG_DIR, writer, detection_property)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "writer.close()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Read back the generated tfrecords and check if the data stored inside the file meets the expectation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Read the files back from the generated tfrecords\n",
+    "def parse_tf_records(file_dir):\n",
+    "    raw_image_dataset = tf.data.TFRecordDataset(file_dir)\n",
+    "\n",
+    "    # Create a dictionary describing the features.\n",
+    "    image_feature_description = {\n",
+    "        'label': tf.io.FixedLenFeature([], tf.int64),\n",
+    "        'image': tf.io.FixedLenFeature([], tf.string),\n",
+    "    }\n",
+    "\n",
+    "    def _parse_image_function(example_proto):\n",
+    "      # Parse the input tf.Example proto using the dictionary above.\n",
+    "      return tf.io.parse_single_example(example_proto, image_feature_description)\n",
+    "\n",
+    "    parsed_image_dataset = raw_image_dataset.map(_parse_image_function)\n",
+    "    \n",
+    "    return parsed_image_dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "parsed_image_dataset = parse_tf_records(TF_FILE_DIR)\n",
+    "\n",
+    "for image_features in parsed_image_dataset:\n",
+    "    print(int(image_features['label']))\n",
+    "    image_raw = image_features['image'].numpy()\n",
+    "    display.display(display.Image(data=image_raw))"
```

Comment:
Similar to above, please compare it with the expected value instead of printing the result if possible.

Reply:
This part is harder to compare. I would like to keep with the original visualization method.

Reply:
Add a comment to note that it's hard to compare.

Reply:
Done!